### PR TITLE
refactor: Rewrite ping

### PR DESCRIPTION
**Pull request by driftluo**
1. rewrite ping
2. remove `last_ping_time`

### CODE REVIEW

**Comment by doitian** [comment]
What's the motivation to rewrite?

**Comment by driftluo** [comment]
> What's the motivation to rewrite?

Reduce unnecessary consumption

File: network/src/protocols/ping.rs [hunk 1]
```diff
@@ -216,36 +210,37 @@ impl ServiceProtocol for PingHandler {
...
-                    .collect::<Vec<PeerId>>()
-                {
-                    self.send_event(Event::Timeout(peer_id));
+                for id in self.connected_session_ids.iter().filter_map(|(id, ps)| {
+                    if ps.processing && ps.elapsed() >= timeout {
+                        Some(id)
+                    } else {
+                        None
+                    }
+                }) {
```
Comment (TheWaWaR) [inline network/src/protocols/ping.rs:237]:
```suggestion
for (id, _ps) in self
    .connected_session_ids
    .iter()
    .filter(|(_id, ps)| ps.processing && ps.elapsed() >= timeout)
{
}
```

Comment (driftluo) [reply network/src/protocols/ping.rs:237]: This is just a little bit of code reduction, the essence has not changed, I have no opinion on this

**Review by quake — APPROVED** [review]

File: network/src/protocols/ping.rs (hunk 1 above)
Comment (doitian) [reply network/src/protocols/ping.rs:237]: :+1: for the suggested change

Comment (driftluo) [reply network/src/protocols/ping.rs]: done

**Review by doitian — APPROVED** [review]

**Review by yangby-cryptape — APPROVED** [review]

**Comment by driftluo** [comment]
bors r=doitian, yangby-cryptape

**Pull request merged**